src/irmin-pack/layers_IO.ml

```diff
@@ -77,3 +77,45 @@ module Unix : S = struct
         let fd = Unix.openfile file Unix.[ O_EXCL; O_RDWR; O_CLOEXEC ] 0o644 in
         { file; fd }
 end
+
+module type WrapS = sig
+  val v : string -> Unix.t Lwt.t
+
+  val close : Unix.t -> unit Lwt.t
+
+  val read_flip : Unix.t -> bool Lwt.t
+
+  val write_flip : bool -> Unix.t -> unit Lwt.t
+
+  val read_generation : Unix.t -> int64 Lwt.t
+
+  val write_generation : int64 -> Unix.t -> unit Lwt.t
+end
+
+module WrapUnix : WrapS = struct
```

Comment:
As you've started doing, it might be interesting to go even further with this and use `Lwt_unix` functions here